src/Optical/Detectors.jl

```diff
@@ -0,0 +1,73 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+import Base
+
+"""
+    AbstractDetector{T<:Number}
```

Comment:
Not sure we need the type in AbstractDetector. For some other cases having the parametric types in the abstract type was important to give the compiler enough information to generate good code. Not sure that will be necessary here.